staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -340,38 +343,29 @@ func (s *SpdyRoundTripper) proxyAuth(proxyURL *url.URL) string {
 // clients may call SpdyRoundTripper.Connection() to retrieve the upgraded
 // connection.
 func (s *SpdyRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	header := utilnet.CloneHeader(req.Header)
-	header.Add(httpstream.HeaderConnection, httpstream.HeaderUpgrade)
-	header.Add(httpstream.HeaderUpgrade, HeaderSpdy31)
-
-	var (
-		conn        net.Conn
-		rawResponse []byte
-		err         error
-	)
+	req = utilnet.CloneRequest(req)
+	req.Header.Add(httpstream.HeaderConnection, httpstream.HeaderUpgrade)
+	req.Header.Add(httpstream.HeaderUpgrade, HeaderSpdy31)
 
-	clone := utilnet.CloneRequest(req)
-	clone.Header = header
-	conn, err = s.Dial(clone)
+	success := false
+	conn, err := s.Dial(req)
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
```

Comment:
This is a future-proof way to not forget that the connection must be closed if an error is returned. Harder to make a mistake when refactoring.

Reply:
I find these constructions with defers hard to think about them, I really don't find this needed, connections are bidirectional and here are wrapped by reader and writer with closers , I see more probabilities of closing something we shouldn't close than solving leaks, bear in mind in http.request and responses is the user responsibility to close the body

Reply:
Removed from this method.